test/cpp/jit/test_class_type.cpp

```diff
@@ -55,5 +56,56 @@ TEST(ClassTypeTest, AddRemoveConstant) {
   ASSERT_TRUE(cls->hasConstant("const3"));
 }
 
+TEST(ClassTypeTest, IdenticalTypesDifferentCus) {
+  auto cu1 = std::make_shared<CompilationUnit>();
+  auto cu2 = std::make_shared<CompilationUnit>();
+
+  // Create two identically named ClassTypes and put them
+  // in separate compilation units.
+  auto cls1 = ClassType::create("foo", cu1);
+  auto cls2 = ClassType::create("foo", cu2);
+
+  // Create a function that accepts "foo" (cls1) as input.
+  Argument arg("arg", cls1);
+  Argument ret("ret", IntType::get());
+
+  FunctionSchema schema("fn", "", {arg}, {ret});
+
+  jit::BuiltinOpFunction method(
+      "method",
+      std::move(schema),
+      [](jit::Stack& stack) mutable -> void {
+        pop(stack);
+        push(stack, 0);
+      },
+      "");
+
+  // Create an object of type cls2.
+  Object obj(cu2, cls2);
+
+  // Call method with the above object; this should
+  // throw an error because the types have identical
+  // names but are in different compilation units.
+  Stack stack;
+  push(stack, obj._ivalue());
+  try {
+    method(stack, {});
+  } catch (const std::exception& e) {
+    // Check that the exception contains the address of the compilation unit
+    // in addition to the ClassType's name.
+    testing::FileCheck()
+        .check("foo(0x")
```

Comment:
Does this test still pass after changing error message? It should be 

```
.check("foo(of python")
.check_same("0x")
```

Reply:
Turns out I did not make the requested changes. 🙃 🔫